### ISSUE TITLE
Help a n00b out! I can't get homebridge to install. 

### ISSUE THREAD

**Issue by sleepyarmistice** [nfarina/homebridge#495]
Trying to install on OS X 10.6. 8 using the command sudo npm install -g homebridge . I'd really so to get this going, though, so could you please (to put it in Reddit-speak) ELI5?

Here's the output that I get from that command and I'm attaching the log file that Terminal indicates I should. Thanks in advance for any help anyone can give!

> curve25519@1.1.0 install /usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/node_modules/curve25519
> node-gyp rebuild

Traceback (most recent call last):
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py", line 18, in <module>
    sys.exit(gyp.script_main())
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/**init**.py", line 533, in script_main
    return main(sys.argv[1:])
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/**init**.py", line 526, in main
    return gyp_main(args)
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/**init**.py", line 502, in gyp_main
    params, options.check, options.circular_check)
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/**init**.py", line 98, in Load
    generator.CalculateVariables(default_variables, params)
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/generator/make.py", line 77, in CalculateVariables
    import gyp.generator.xcode as xcode_generator
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/generator/xcode.py", line 8, in <module>
    import gyp.xcode_ninja
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/xcode_ninja.py", line 19, in <module>
    import xml.sax.saxutils
  File "/System/Library/Frameworks/Python.framework/Versions/2.6/lib/python2.6/xml/sax/saxutils.py", line 6, in <module>
    import os, urlparse, urllib, types
  File "/System/Library/Frameworks/Python.framework/Versions/2.6/lib/python2.6/urllib.py", line 1347, in <module>
    from _scproxy import _get_proxy_settings, _get_proxies
ImportError: No module named _scproxy
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:355:16)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 10.8.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/node_modules/curve25519
gyp ERR! node -v v4.2.6
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok 

> bignum@0.11.0 install /usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/node_modules/srp/node_modules/bignum
> node-gyp configure build

Traceback (most recent call last):
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py", line 18, in <module>
    sys.exit(gyp.script_main())
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/**init**.py", line 533, in script_main
    return main(sys.argv[1:])
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/**init**.py", line 526, in main
    return gyp_main(args)
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/**init**.py", line 502, in gyp_main
    params, options.check, options.circular_check)
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/**init**.py", line 98, in Load
    generator.CalculateVariables(default_variables, params)
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/generator/make.py", line 77, in CalculateVariables
    import gyp.generator.xcode as xcode_generator
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/generator/xcode.py", line 8, in <module>
    import gyp.xcode_ninja
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/xcode_ninja.py", line 19, in <module>
    import xml.sax.saxutils
  File "/System/Library/Frameworks/Python.framework/Versions/2.6/lib/python2.6/xml/sax/saxutils.py", line 6, in <module>
    import os, urlparse, urllib, types
  File "/System/Library/Frameworks/Python.framework/Versions/2.6/lib/python2.6/urllib.py", line 1347, in <module>
    from _scproxy import _get_proxy_settings, _get_proxies
ImportError: No module named _scproxy
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:355:16)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 10.8.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "configure" "build"
gyp ERR! cwd /usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/node_modules/srp/node_modules/bignum
gyp ERR! node -v v4.2.6
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok 

> ed25519@0.0.3 install /usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/node_modules/ed25519
> node-gyp rebuild

Traceback (most recent call last):
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py", line 18, in <module>
    sys.exit(gyp.script_main())
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/**init**.py", line 533, in script_main
    return main(sys.argv[1:])
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/**init**.py", line 526, in main
    return gyp_main(args)
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/**init**.py", line 502, in gyp_main
    params, options.check, options.circular_check)
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/**init**.py", line 98, in Load
    generator.CalculateVariables(default_variables, params)
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/generator/make.py", line 77, in CalculateVariables
    import gyp.generator.xcode as xcode_generator
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/generator/xcode.py", line 8, in <module>
    import gyp.xcode_ninja
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/xcode_ninja.py", line 19, in <module>
    import xml.sax.saxutils
  File "/System/Library/Frameworks/Python.framework/Versions/2.6/lib/python2.6/xml/sax/saxutils.py", line 6, in <module>
    import os, urlparse, urllib, types
  File "/System/Library/Frameworks/Python.framework/Versions/2.6/lib/python2.6/urllib.py", line 1347, in <module>
    from _scproxy import _get_proxy_settings, _get_proxies
ImportError: No module named _scproxy
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:355:16)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 10.8.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/node_modules/ed25519
gyp ERR! node -v v4.2.6
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok 

> mdns@2.2.11 install /usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/node_modules/mdns
> node-gyp rebuild

Traceback (most recent call last):
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py", line 18, in <module>
    sys.exit(gyp.script_main())
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/**init**.py", line 533, in script_main
    return main(sys.argv[1:])
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/**init**.py", line 526, in main
    return gyp_main(args)
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/**init**.py", line 502, in gyp_main
    params, options.check, options.circular_check)
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/**init**.py", line 98, in Load
    generator.CalculateVariables(default_variables, params)
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/generator/make.py", line 77, in CalculateVariables
    import gyp.generator.xcode as xcode_generator
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/generator/xcode.py", line 8, in <module>
    import gyp.xcode_ninja
  File "/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/xcode_ninja.py", line 19, in <module>
    import xml.sax.saxutils
  File "/System/Library/Frameworks/Python.framework/Versions/2.6/lib/python2.6/xml/sax/saxutils.py", line 6, in <module>
    import os, urlparse, urllib, types
  File "/System/Library/Frameworks/Python.framework/Versions/2.6/lib/python2.6/urllib.py", line 1347, in <module>
    from _scproxy import _get_proxy_settings, _get_proxies
ImportError: No module named _scproxy
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:355:16)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 10.8.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/node_modules/mdns
gyp ERR! node -v v4.2.6
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok 
npm ERR! Darwin 10.8.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "-g" "homebridge"
npm ERR! node v4.2.6
npm ERR! npm  v2.14.12
npm ERR! code ELIFECYCLE

npm ERR! curve25519@1.1.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the curve25519@1.1.0 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the curve25519 package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls curve25519
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/xxxx/npm-debug.log

**Comment by tktino** [comment]
i'm on ubuntu, which isn't too far off from osx. here's how i get things started in the matter of mins (in order)


sudo apt-get install libavahi-compat-libdnssd-dev
sudo apt-get install nodejs
sudo apt-get install npm
sudo apt-get install nodejs-legacy
sudo apt-get install git
sudo npm -g install npm@latest


sudo npm install -g homebridge

**Comment by sleepyarmistice** [comment]
Sadly that didn't seem to work for me. You can't use apt-get on OS X, but even after I figured out how to work around that I'm still in the same place I started.

**Comment by tktino** [comment]
The apt-get command is part of the Advanced Packaging Tools for Debian Linux and is not supported on OS X. It's possible that if this had worked in the past you aliased another command.

If you are looking for a command line tool for installing and managing packages on OS X then you are looking for Homebrew. It is easy to install via the command line using this command

ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
Once installed you can use the brew command much like apt-get.

**Comment by AppleTechy** [comment]
Were you able to get it installed?

**Comment by sleepyarmistice** [comment]
No. As I recall, I concluded that the necessary dependencies weren't available for Snow Leopard. I ended up getting a Raspberry Pi, which runs homebridge nicely.

**Comment by tktino** [comment]
Very nice! I'm running homebridge on a RPi Zero!

Sent from Outlook for iPhone




On Wed, Mar 16, 2016 at 9:55 PM -0700, "sleepyarmistice" <notifications@github.com> wrote:










No. As I recall, I concluded that the necessary dependencies weren't available for Snow Leopard. I ended up getting a Raspberry Pi, which runs homebridge nicely.
